app/controllers/indicators_controller.rb

```diff
@@ -72,6 +73,16 @@ def set_indicator_data
     @summary_value = calculations[:summary_value]
   end
 
+  def set_coin_stats
+    @coin_stats = Rails.cache.read("indicators/#{@coin.slug}:stats")
```

Comment:
Sorry for nitpicking but the best practice is to use `Rails.cache.fetch` and pass in the block that represents what happens upon cache miss.

https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch

Reply:
Just to be explicit, this way of operating is important because it lazily populates the cache for subsequent requests, whereas in your current implementation it is possible for the cache to never be populated nor read.